### PR TITLE
Add pbc_wrap_batched_and_get_lattice_shifts to handle unwrapped positions in nl

### PR DESCRIPTION
**Pull request by CompRhys**
Addresses https://github.com/TorchSim/torch-sim/issues/437

### CODE REVIEW

File: torch_sim/transforms.py [hunk 1]
```diff
@@ -139,43 +139,68 @@ def pbc_wrap_batched(
...
+    pbc_mask = pbc_per_atom & active_per_atom
+
+    int_shifts = torch.where(pbc_mask, torch.floor(frac), torch.zeros_like(frac))
+    wrapped_frac = frac - int_shifts
+
+    cell_per_atom = cell_T[system_idx]
+    wrapped_pos = torch.bmm(cell_per_atom, wrapped_frac.unsqueeze(2)).squeeze(2)
+    out = torch.where(active_per_atom, wrapped_pos, positions)
+    shifts = torch.where(active_per_atom, int_shifts, torch.zeros_like(int_shifts))
+    return out, shifts
```
Comment (CompRhys) [inline torch_sim/transforms.py:210]: key idea was just to keep these extra shifts around and adjust the nl shifts accordingly.

**Comment by CompRhys** [comment]
alchemiops can run on CPU but many of our test cases hit it's default max neighbor amounts

File: tests/test_fix_symmetry.py [hunk 2]
```diff
@@ -34,7 +34,7 @@ def make_structure(name: str, repeats: int = REPEATS) -> Atoms:
         "fcc": lambda: bulk("Cu", "fcc", a=3.6),
         "hcp": lambda: bulk("Ti", "hcp", a=2.95, c=4.68),
         "diamond": lambda: bulk("Si", "diamond", a=5.43),
-        "bcc": lambda: bulk("Al", "bcc", a=2 / np.sqrt(3), cubic=True),
```
Comment (CompRhys) [inline tests/test_fix_symmetry.py:37]: too dense lead to nbr explosion exceeding alchemiops padding heuristic

File: torch_sim/models/interface.py [hunk 3]
```diff
@@ -371,3 +376,34 @@ def validate_model_outputs(  # noqa: C901, PLR0915
         raise ValueError(f"{fe_model_output['forces'].shape=} != (1, 3)")
     if stress_computed and fe_model_output["stress"].shape != (1, 3, 3):
         raise ValueError(f"{fe_model_output['stress'].shape=} != (1, 3, 3)")
+
+    # Translating one atom by a full lattice vector should not change outputs.
```
Comment (CompRhys) [inline torch_sim/models/interface.py:380]: @alphalm4 This test seemingly passes with the upstream sevennet so I would guess that no additional changes are needed for the precision issue you mentioned at least at the 0.1 meV level (1e-4)

Comment (alphalm4) [reply torch_sim/models/interface.py:380]: Great! I think the validation condition should be stricter in this [line](https://github.com/TorchSim/torch-sim/blob/a89ebe65fbb05be0a19aefccaf0d4d6b9e986061/torch_sim/models/interface.py#L384) like below,
since I found only using `+ lattice_vec` luckly PASSES even though NL things are not patched.

`shifted_state.positions[0] = shifted_state.positions[0] + 10 * lattice_vec`

I also checked that the current PR version, together with `nvalchemi-toolkit-ops==0.3.0`  and `vesin[torch]==0.5.3` passes all sevenn+torchsim tests with all four `torchsim_nl`s. Thanks for considering this.
